src/hercule.coffee

```diff
@@ -3,6 +3,7 @@ blanket = require 'blanket' if process.env.COVERAGE
 path = require 'path'
 _ = require 'lodash'
 utils = require './utils'
+validUrlRegex = /(http|https):\/\/(\w+:{0,1}\w*)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%!\-\/]))?/
```

Comment:
I'm happy to investigate extending the grammar to identify a URL and provide the correct `link.type`.